python/pycrdt/_transaction.py

```diff
@@ -75,9 +99,6 @@ def __exit__(
             assert self._txn is not None
             if not isinstance(self, ReadTransaction):
                 self._txn.commit()
-                origin_hash = self._txn.origin()
-                if origin_hash is not None:
-                    del self._doc._origins[origin_hash]
```

Comment:
That was a very deterministic behavior, why did you move that part to the transaction's `__del__` method?

Reply:
> If you leave the logic there, then the tests won't work. Because when requesting the transaction origin from the event object the entry in the `doc._origins` will be gone.

I think it's just a matter of [dropping the transaction](https://github.com/jupyter-server/pycrdt/blob/3b6c0bb137bde91c7f729e2b50c72794fcfcec7e/python/pycrdt/_transaction.py#L83) after removing the origin from the doc.

> The usage of `__del__` is not less deterministic

Yes it is, because now [tests behave differently if running on CPython or PyPy](https://github.com/jupyter-server/pycrdt/blob/9fed42e8b54e3cfdbbddf56ee4265c8b53b8c33d/tests/test_transaction.py#L105).